### ISSUE TITLE
Bloated buffer result with JOIN_MITRE

### ISSUE THREAD

**Issue by mggevaert** [locationtech/jts#847]
Greetings, 

I noticed ugly results when using the option "JOIN_MITRE" on following geometry:
![image](https://user-images.githubusercontent.com/72023004/156419925-741ce43f-290e-4db7-9376-fac392a342f2.png)

```java
        BufferParameters bufferParameters = new BufferParameters();
        bufferParameters.setJoinStyle(BufferParameters.JOIN_MITRE);
        Polygon inflatedGeometry = (Polygon) BufferOp.bufferOp(geometry, 0.5, bufferParameters);
```
Geometry is a polygon with points:
`{2, 0}, {0, 1.5}, {1, 1.5}, {1, 3}, {0, 2.3}, {2, 4}, {2, 0}`

Is this expected behavior? I'm looking for behavior similar to "JOIN_BEVEL", but that still respects the minimum distance at all points
![image](https://user-images.githubusercontent.com/72023004/156420843-87fb94bf-5147-454f-9ef1-5e2a5a6ab240.png)

**Comment by dr-jts** [comment]
What version of JTS?  Trying this with the current codebase looks fine to me:
```
POLYGON ((2 0, 0 1.5, 1 1.5, 1 3, 0 2.3, 2 4, 2 0))
```
![image](https://user-images.githubusercontent.com/3529053/156422567-31ec34f8-7307-4eb3-9bb9-c117bf4c6aaf.png)

**Comment by mggevaert** [comment]
Hi dr-jts,

Thank you for your quick response,
I discovered this issue on 16.0 and upgraded to 18.2 to verify it still exists.

Based on your message I suspect the result I got, (except for the lines going to the inner triangle) is expected result.
My concern was mostly with the outer shell.

Using a mitre limit of `1` solves the issue for me since I need the buffer functionality to find borders around polygons that are always at least "x"  amount of distance away from the polygon.
![image](https://user-images.githubusercontent.com/72023004/156426222-948abe28-9608-4de8-b7a7-d1a2d74e3141.png)

My problem is solved, I will leave this issue open in case this is something which you would like to investigate further.

Thanks for the help!
